app/controllers/submissions_controller.rb

```diff
@@ -971,13 +971,17 @@ def notebook_to_html(file_contents, unique_path, type)
           '--template', 'markus-html-template'
         ]
       end
-      file_contents = JSON.parse(file_contents)
-      if file_contents['metadata'].key?('widgets')
-        file_contents['metadata'].delete('widgets')
+      if !SubmissionFile.is_binary?(file_contents)
+        file_contents = JSON.parse(file_contents)
+        if file_contents['metadata'].key?('widgets')
+          file_contents['metadata'].delete('widgets')
+        end
+        file_contents = JSON.generate(file_contents)
+        _stdout, stderr, status = Open3.capture3(*args, stdin_data: file_contents)
+        return "#{I18n.t('submissions.cannot_display')}<br/><br/>#{stderr.lines.last}" unless status.exitstatus.zero?
+      else
+        return "#{I18n.t('submissions.file_extension_mismatch')}<br/>"
```

Comment:
I don't think the trailing `<br/>` is necessary, unless you have a screenshot to show what the UI looks like without it?